Anlab.Mvc/Services/IOrderService.cs

```diff
@@ -121,6 +123,43 @@ public void PopulateOrderWithLabDetails(OrderSaveModel model, Order orderToUpdat
             orderToUpdate.SaveDetails(orderDetails);
         }
 
+        public async Task OverwiteOrderWithTestsCompleted(Order orderToUpdate)
+        {
+            if (string.IsNullOrWhiteSpace(orderToUpdate.OrderRequest))
+            {
+                throw new Exception("OrderRequest not populated"); //TODO: Something better
+            }
+            var testCodes = await _labworksService.GetTestCodesCompletedForOrder(orderToUpdate.OrderRequest);
+
+            var testIds = _context.TestItems.Where(a => testCodes.Contains(a.Code)).Select(s => s.Id).ToArray();
+            var tests = await PopulateSelectedTestsItemModel(testIds);
+
+            var orderDetails = orderToUpdate.GetOrderDetails();
+            var calcualtedTests = new List<TestDetails>();
+            foreach (var test in tests)
+            {
+                var cost = orderDetails.Payment.IsInternalClient ? test.InternalCost : test.ExternalCost;
+                var costAndQuantity = cost * orderDetails.Quantity;
+
+                calcualtedTests.Add(new TestDetails

```

Comment:
@srkirkland Yeah. There are enough differences that I want to think about it some more.